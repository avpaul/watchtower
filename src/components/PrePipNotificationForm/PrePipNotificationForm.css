.prepipOutline {
  background-color: #ffffff;
  box-sizing: border-box;
  color: #74787e;
  padding: 43px;
  font-family: Avenir, Helvetica, sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  margin-top: 0;
  text-align: left;
}

.prepipBanner1 {
  box-sizing: border-box;
  padding: 25px;
  display: inline-block;
  color: #0459e4;
  font-size: 19px;
  font-weight: bold;
  text-decoration: none;
  text-align: left;
  vertical-align: middle;
  background-color: white;
  padding-left: 10px;
}

.prepipBanner {
  box-sizing: border-box;
  padding-left: 15%;
  display: inline-block;
  color: #0459e4;
  font-size: 19px;
  font-weight: bold;
  padding: 25px;
  background-color: white;
  text-align: right;
}

.prepipTable th,
td,
thead {
  border: 1px solid #d1d1d1;
  box-sizing: border-box;
  font-size: 15px;
  line-height: 18px;
  padding-right: 20px;
  color: rgba(0, 0, 0, 0.65);
  border-right: 1px solid #e8e8e8;
  border-bottom: 1px solid #e8e8e8;
  padding: 8px;
  margin: 0;
  text-align: left;
  min-width: 120px;
  border-collapse: separate;
  border-spacing: 2px;
}

.tableHeader tr,
th {
  border: 1px solid #d1d1d1;
  box-sizing: border-box;
  padding: 8px;
  margin: 0;
  padding-right: 20px;
  background-color: #fafafa;
  color: rgba(0, 0, 0, 0.85);
  border-right: 1px solid #e8e8e8;
  border-bottom: 1px solid #e8e8e8;
  text-align: left;
  border-collapse: separate;
  border-spacing: 2px;
}

.prepipHtml {
  display: block;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
}

.feedbackTable {
  width: 92vw;
  margin: auto;
}

.feedbackTable td {
  border: none;
  border-bottom: 1px solid #0459e4;
}

.btn-print-prepip-form {
  box-sizing: border-box;
  background-color: #d1d1d1;
  color: #ffffff;
  border-radius: 4px;
  margin-top: 15px;
  font-size: 14px;
  font-weight: bold;
}

.btn-print-prepip-form:hover {
  background-color: blue;
}
