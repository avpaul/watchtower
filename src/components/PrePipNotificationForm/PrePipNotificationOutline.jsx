import React from 'react';
import PropTypes from 'prop-types';
import getFeedbackIndex from '../../utils/formatFeedbackInstance';
import LogoSection from '../Header/LogoSection';

export const managerName = data => {
  if (data.Manager === 'null') {
    return null;
  }
  if (typeof data.Manager === 'string') {
    return `${JSON.parse(data.Manager).name}, Manager`;
  }
  return `${data.Manager.name}, Manager`;
};

const PrePipNotificationOutline = ({ fellowFeedback, serialNumber }) => (
  <div className="prepipForm" id="prepipForm">
    <table className="feedbackTable">
      <tr>
        <td>
          <div className="prepipBanner1">
            <LogoSection />
          </div>
        </td>
        <td colSpan="6">
          <div className="prepipBanner">Documented Feedback</div>
        </td>
      </tr>
    </table>
    <p className="prepipHtml">
      Dear <b>{fellowFeedback && `${fellowFeedback.name}`},</b>
    </p>
    <p className="prepipHtml">
      The purpose of this email is to notify you that your performance has been
      and is currently not at a satisfactory level. As such, you are required to
      level up on the following skills/areas within the coming weeks:
    </p>

    <table className="tableRing">
      <tr>
        <td>
          <table className="prepipTable">
            <thead className="tableHeader">
              <tr>
                <th>S/N</th>
                <th>
                  Performance Issue Area(s) (Dev Pulse Attribute/LMSOutput)
                </th>
                <th>Rolling Average</th>
                <th>Context</th>
                <th>Areas of Improvement</th>
              </tr>
            </thead>
            <tbody>
              <tr className="dashboard-table__row">
                <td>
                  <b>{fellowFeedback && serialNumber}</b>
                </td>
                <td>
                  <b className="text-capitalize">
                    {fellowFeedback && fellowFeedback.Criteria}
                  </b>
                </td>
                <td>
                  <b>{fellowFeedback && fellowFeedback.Rating}</b>
                </td>
                <td>
                  <b>{fellowFeedback && fellowFeedback.Context}</b>
                </td>
                <td>
                  <b>{fellowFeedback && fellowFeedback.Recommendation}</b>
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </table>

    <p className="prepipHtml">
      Please note that this is the{' '}
      {fellowFeedback &&
        `${getFeedbackIndex(
          fellowFeedback.Instances,
          fellowFeedback.startDate
        )} `}
      feedback you have received since the performance issue was observed. In
      {/* eslint-disable-next-line */}
      order to meet the Program advancement standard of >=1 on all DevPulse
      attributes, you may need to exceed expectation in these areas within the
      coming weeks.
    </p>
    <p className="prepipHtml">
      We encourage you to reach out to your TTL who is available to give you
      feedback and guidance. You can also seek help from your colleagues or
      cohort rep or any member of the fellowâ€™s council or leadership team as
      needed as you strive to improve on these issue areas.
    </p>
    <p className="prepipHtml">TIA, </p>
    <p>
      <b>{managerName(fellowFeedback)}</b>
    </p>
  </div>
);

PrePipNotificationOutline.propTypes = {
  serialNumber: PropTypes.number.isRequired,
  fellowFeedback: PropTypes.shape({}).isRequired
};
export default PrePipNotificationOutline;
