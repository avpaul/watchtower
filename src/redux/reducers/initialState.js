import { ALL } from '../constants/fellowFilters';

export const INDEX_PAGE_SIZE_DEFAULT = 20;
export const INDEX_PAGE_SIZE_OPTIONS = ['5', '10', '19', '30', '50', '100'];

const genericAPIStateArray = {
  loading: false,
  data: [],
  error: null
};

const genericAPIStateObject = {
  loading: false,
  data: {},
  error: null
};

export default {
  emsDashboard: {
    fellowsSummary: {
      loading: false,
      summary: [],
      error: ''
    }
  },
  fellows: {
    pagination: {
      perPage: 25,
      page: 1,
      pages: 1,
      firstPageURL: '',
      nextPageURL: '',
      prevPageURL: '',
      finalPageURL: ''
    },
    fellows: [],
    summary: { fellowsofftrack: 0, gteWk5OffTrack: 0, fellowsontrack: 0 },
    loading: false,
    error: ''
  },
  visibilityFilter: ALL,
  fellowCountHistory: {
    countSummary: {},
    loading: false
  },
  fellowsProgress: {
    loading: false,
    data: {
      D0A: [],
      D0B: [],
      D0: []
    },
    error: null
  },
  fellowDevPulse: {
    loading: false,
    ratings: [],
    averageRatings: {
      communication: 'N/A',
      integration: 'N/A',
      quantity: 'N/A',
      quality: 'N/A',
      professionalism: 'N/A',
      initiative: 'N/A'
    },
    error: null
  },
  fellowLmsSubmissions: {
    loading: false,
    lmsSubmissions: {},
    error: null
  },
  manager: {
    loading: false,
    data: {},
    error: null
  },
  opsSummary: {
    data: {
      managers: {
        ttls: [],
        lfs: [],
        averageFellowsPerTtl: 0,
        averageFellowsPerLf: 0
      },
      fellowsCount: {
        Total: 0
      },
      locations: []
    },
    loading: false
  },
  managerFellows: {
    loading: false,
    managerFellows: {},
    error: ''
  },
  fellow: {
    loading: false,
    fellow: {},
    error: ''
  },
  d1Fellow: {
    loading: false,
    fellow: {},
    error: ''
  },
  notification: {
    loading: false,
    notification: [],
    error: ''
  },
  unreadnotification: {
    loading: false,
    unreadnotification: {},
    error: ''
  },
  readnotification: {
    loading: false,
    readnotification: {},
    error: ''
  },
  fellowsSummary: {
    loading: false,
    fellowsSummaryToday: {},
    fellowsSummaryTrend: {},
    data: {},
    error: ''
  },
  engineeringManagerSimsLeads: {
    loading: false,
    data: {
      managers: {},
      locations: []
    },
    error: null
  },
  updateManagerNotification: {
    loading: false,
    managerNotification: [],
    error: null
  },
  managerFellowsSummary: {
    loading: false,
    data: [],
    error: null
  },
  feedback: {
    loading: false,
    data: [],
    error: null
  },
  fellowFeedback: {
    feedback: [],
    fellowFeedback: {}
  },
  pipData: {
    data: {},
    loading: false,
    error: null
  },
  performanceData: {
    loading: false,
    data: {
      today: {},
      trend: {}
    },
    error: ''
  },
  fellowPipFeedback: {
    loading: false,
    feedback: [],
    error: null
  },
  fellowPrePipFeedback: {
    loading: false,
    feedback: [],
    error: null
  },
  pipDeactivation: {
    success: null
  },
  createProject: genericAPIStateObject,
  createProjectVacancies: genericAPIStateObject,
  allProjects: genericAPIStateArray,
  fetchProjectManagers: genericAPIStateArray,
  fetchProjectTechnologies: genericAPIStateArray,
  editSingleProject: genericAPIStateObject,
  singleProject: genericAPIStateObject,
  createCertificactionVacancies: genericAPIStateObject,
  addProjectManager: {
    manager: {}
  },
  addProjectTechnology: {
    technology: ''
  },
  fetchSlackChannels: genericAPIStateArray,
  cadreEngineers: {
    loading: false,
    cadreEngineers: [],
    error: ''
  },
  cadreVacancies: genericAPIStateObject,
  reports: {
    loading: false,
    error: '',
    data: [],
    meta: {
      page: 1,
      perPage: INDEX_PAGE_SIZE_DEFAULT,
      perPageOptions: INDEX_PAGE_SIZE_OPTIONS,
      total: 0
    }
  },
  allRoles: genericAPIStateArray,
  fetchAllRoles: genericAPIStateArray,
  fetchActiveRole: genericAPIStateArray,
  createRole: genericAPIStateArray,
  roleSkills: genericAPIStateArray,
  getAllVacancies: genericAPIStateObject,
  allCetifications: genericAPIStateArray,
  editCertification: genericAPIStateObject,
  certifications: genericAPIStateArray,
  getCertification: genericAPIStateObject,
  editProjectVacancies: genericAPIStateObject,
  deleteProjectVacancies: genericAPIStateObject,
  projectVacanciesOnFocus: {},
  deleteCertification: genericAPIStateObject,
  certificationOnFocus: {},
  certificationApplicants: genericAPIStateArray,
  certificationApplication: genericAPIStateObject
};
