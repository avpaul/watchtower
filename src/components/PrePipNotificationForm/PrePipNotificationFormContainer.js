import { connect } from 'react-redux';
import PrePipNotificationForm from './PrePipNotificationForm';

const mapStateToProps = state => ({
  fellowFeedback: state.fellowFeedback
});

export default connect(mapStateToProps)(PrePipNotificationForm);
