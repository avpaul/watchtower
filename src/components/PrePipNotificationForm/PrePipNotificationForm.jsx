import React, { Component } from 'react';
import PropTypes from 'prop-types';
import html2canvas from 'html2canvas';
import JsPdf from 'jspdf';
import PrePipNotificationOutline from './PrePipNotificationOutline';
import './PrePipNotificationForm.scss';

class PrePipNotificationForm extends Component {
  constructor() {
    super();
    this.input = React.createRef();
  }

  printDocument = () => {
    const input = this.input.current;
    html2canvas(input, { scale: '2' }).then(canvas => {
      const imgData = canvas.toDataURL('image/png');
      this.createPdf(imgData);
    });
  };

  createPdf = imgData => {
    const pdf = new JsPdf({ orientation: 'p', unit: 'mm' });
    pdf.internal.scaleFactor = 1;
    pdf.addImage(imgData, 'JPEG', 5, 10, 200, 150);
    pdf.save('download.pdf');
  };

  render() {
    const { fellowFeedback, history } = this.props;
    if (fellowFeedback.fellowFeedback)
      if (Object.keys(fellowFeedback.fellowFeedback).length === 0)
        history.push('/feedback');
    return (
      <div className="prepipOutline">
        <div className="feedbackDashboard container-fluid">
          <div className="row" ref={this.input}>
            <PrePipNotificationOutline
              fellowFeedback={fellowFeedback}
              serialNumber={parseInt(fellowFeedback.index, 10) + 1}
            />
          </div>
        </div>
        <div>
          <button
            type="button"
            className="btn-print-prepip-form"
            onClick={this.printDocument}
          >
            Download as PDF
          </button>
        </div>
      </div>
    );
  }
}

PrePipNotificationForm.propTypes = {
  fellowFeedback: PropTypes.shape({}),
  history: PropTypes.shape({}).isRequired
};

PrePipNotificationForm.defaultProps = {
  fellowFeedback: undefined
};
export default PrePipNotificationForm;
