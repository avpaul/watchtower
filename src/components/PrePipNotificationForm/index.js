import PrePipNotificationFormContainer from './PrePipNotificationFormContainer';

export default PrePipNotificationFormContainer;
