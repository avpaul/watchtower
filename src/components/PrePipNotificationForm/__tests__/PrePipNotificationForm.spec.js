import React from 'react';
import jspdf from 'jspdf';
import html2canvas from 'html2canvas';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import initialState from '../../../redux/reducers/initialState';
import PrePipNotificationForm from '../PrePipNotificationForm';
import PrePipNotificationFormContainer from '../PrePipNotificationFormContainer';

jest.mock('html2canvas', () =>
  jest.fn().mockImplementation(() =>
    Promise.resolve({
      toDataURL: () => {}
    })
  )
);

jest.mock('jspdf', () =>
  jest.fn().mockImplementation(() => ({
    internal: { pageSize: { getWidth: () => 2 } },
    addImage: () => jest.fn(),
    save: () => {}
  }))
);

describe('Pre pip notification form', () => {
  const mockStore = configureStore();
  let store;
  let wrapper;
  let containerWrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    const props = {
      fellowFeedback: {
        fellowFeedback: {},
        Attribute: null,
        Context: 'Hi Sinmiloluwa',
        Criteria: 'lms',
        name: 'Oloyede Sinmiloluwa',
        Manager: {
          name: 'Trust Birungi',
          email: 'trust.birungi@andela.com'
        },
        Recommendation: null,
        index: '0'
      },
      history: []
    };

    wrapper = shallow(<PrePipNotificationForm store={store} {...props} />);
    containerWrapper = shallow(
      <PrePipNotificationFormContainer store={store} />
    );
  });

  it('renders to match shapshot', () => {
    expect(wrapper).toMatchSnapshot();
    expect(containerWrapper).toMatchSnapshot();
  });

  it('renders details as expected', () => {
    wrapper.instance().printDocument();
    wrapper.instance().createPdf('0x64');
    expect(html2canvas).toHaveBeenCalled();
    expect(jspdf).toHaveBeenCalledWith({ orientation: 'p', unit: 'mm' });
  });
});
