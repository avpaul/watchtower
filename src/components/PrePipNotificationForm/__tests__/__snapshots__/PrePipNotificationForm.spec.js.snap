// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pre pip notification form renders to match shapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrePipNotificationForm
    fellowFeedback={
        Object {
            "Attribute": null,
            "Context": "Hi Sinmiloluwa",
            "Criteria": "lms",
            "Manager": Object {
              "email": "trust.birungi@andela.com",
              "name": "Trust Birungi",
            },
            "Recommendation": null,
            "fellowFeedback": Object {},
            "index": "0",
            "name": "Sinmiloluwa Oloyede",
          }
    }
    history={
        Array [
            "/feedback",
          ]
    }
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="feedbackDashboard container-fluid"
>
          <div
                    className="row"
          >
                    <PrePipNotificationOutline
                              fellowFeedback={
                                        Object {
                                                  "Attribute": null,
                                                  "Context": "Hi Sinmiloluwa",
                                                  "Criteria": "lms",
                                                  "Manager": Object {
                                                    "email": "trust.birungi@andela.com",
                                                    "name": "Trust Birungi",
                                                  },
                                                  "Recommendation": null,
                                                  "fellowFeedback": Object {},
                                                  "index": "0",
                                                  "name": "Sinmiloluwa Oloyede",
                                                }
                              }
                              serialNumber={1}
                    />
          </div>
</div>,
        <div>
          <button
                    className="btn-print-prepip-form"
                    onClick={[Function]}
                    type="button"
          >
                    Download as PDF
          </button>
</div>,
      ],
      "className": "prepipOutline",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
>
            <PrePipNotificationOutline
                        fellowFeedback={
                                    Object {
                                                "Attribute": null,
                                                "Context": "Hi Sinmiloluwa",
                                                "Criteria": "lms",
                                                "Manager": Object {
                                                  "email": "trust.birungi@andela.com",
                                                  "name": "Trust Birungi",
                                                },
                                                "Recommendation": null,
                                                "fellowFeedback": Object {},
                                                "index": "0",
                                                "name": "Sinmiloluwa Oloyede",
                                              }
                        }
                        serialNumber={1}
            />
</div>,
          "className": "feedbackDashboard container-fluid",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PrePipNotificationOutline
              fellowFeedback={
                            Object {
                                          "Attribute": null,
                                          "Context": "Hi Sinmiloluwa",
                                          "Criteria": "lms",
                                          "Manager": Object {
                                            "email": "trust.birungi@andela.com",
                                            "name": "Trust Birungi",
                                          },
                                          "Recommendation": null,
                                          "fellowFeedback": Object {},
                                          "index": "0",
                                          "name": "Sinmiloluwa Oloyede",
                                        }
              }
              serialNumber={1}
/>,
            "className": "row",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "fellowFeedback": Object {
                "Attribute": null,
                "Context": "Hi Sinmiloluwa",
                "Criteria": "lms",
                "Manager": Object {
                  "email": "trust.birungi@andela.com",
                  "name": "Trust Birungi",
                },
                "Recommendation": null,
                "fellowFeedback": Object {},
                "index": "0",
                "name": "Sinmiloluwa Oloyede",
              },
              "serialNumber": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn-print-prepip-form"
            onClick={[Function]}
            type="button"
>
            Download as PDF
</button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Download as PDF",
            "className": "btn-print-prepip-form",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Download as PDF",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="feedbackDashboard container-fluid"
>
            <div
                        className="row"
            >
                        <PrePipNotificationOutline
                                    fellowFeedback={
                                                Object {
                                                            "Attribute": null,
                                                            "Context": "Hi Sinmiloluwa",
                                                            "Criteria": "lms",
                                                            "Manager": Object {
                                                              "email": "trust.birungi@andela.com",
                                                              "name": "Trust Birungi",
                                                            },
                                                            "Recommendation": null,
                                                            "fellowFeedback": Object {},
                                                            "index": "0",
                                                            "name": "Sinmiloluwa Oloyede",
                                                          }
                                    }
                                    serialNumber={1}
                        />
            </div>
</div>,
          <div>
            <button
                        className="btn-print-prepip-form"
                        onClick={[Function]}
                        type="button"
            >
                        Download as PDF
            </button>
</div>,
        ],
        "className": "prepipOutline",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
>
              <PrePipNotificationOutline
                            fellowFeedback={
                                          Object {
                                                        "Attribute": null,
                                                        "Context": "Hi Sinmiloluwa",
                                                        "Criteria": "lms",
                                                        "Manager": Object {
                                                          "email": "trust.birungi@andela.com",
                                                          "name": "Trust Birungi",
                                                        },
                                                        "Recommendation": null,
                                                        "fellowFeedback": Object {},
                                                        "index": "0",
                                                        "name": "Sinmiloluwa Oloyede",
                                                      }
                            }
                            serialNumber={1}
              />
</div>,
            "className": "feedbackDashboard container-fluid",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <PrePipNotificationOutline
                fellowFeedback={
                                Object {
                                                "Attribute": null,
                                                "Context": "Hi Sinmiloluwa",
                                                "Criteria": "lms",
                                                "Manager": Object {
                                                  "email": "trust.birungi@andela.com",
                                                  "name": "Trust Birungi",
                                                },
                                                "Recommendation": null,
                                                "fellowFeedback": Object {},
                                                "index": "0",
                                                "name": "Sinmiloluwa Oloyede",
                                              }
                }
                serialNumber={1}
/>,
              "className": "row",
            },
            "ref": Object {
              "current": null,
            },
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "fellowFeedback": Object {
                  "Attribute": null,
                  "Context": "Hi Sinmiloluwa",
                  "Criteria": "lms",
                  "Manager": Object {
                    "email": "trust.birungi@andela.com",
                    "name": "Trust Birungi",
                  },
                  "Recommendation": null,
                  "fellowFeedback": Object {},
                  "index": "0",
                  "name": "Sinmiloluwa Oloyede",
                },
                "serialNumber": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn-print-prepip-form"
              onClick={[Function]}
              type="button"
>
              Download as PDF
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Download as PDF",
              "className": "btn-print-prepip-form",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Download as PDF",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Pre pip notification form renders to match shapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(PrePipNotificationForm)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "fellowFeedback": Object {
        "feedback": Array [],
        "fellowFeedback": Object {},
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "fellowFeedback": Object {
          "feedback": Array [],
          "fellowFeedback": Object {},
        },
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": undefined,
  },
}
`;
